Sources/SwiftSyntaxMacroExpansion/MacroSystem.swift

```diff
@@ -510,6 +537,56 @@ extension MacroApplication {
     )
   }
 
+  private func expandPreamble(
+    for decl: some DeclSyntaxProtocol & HasTrailingOptionalCodeBlock
+  ) -> [CodeBlockItemSyntax] {
+    var preamble: [CodeBlockItemSyntax] = []
+    let macroAttributes = getMacroAttributes(attachedTo: DeclSyntax(decl), ofType: PreambleMacro.Type.self)
+    for (attribute, preambleMacro) in macroAttributes {
+      do {
+        let newPreamble = try preambleMacro.expansion(
+          of: attribute,
+          providingPreambleFor: decl,
+          in: context
+        )
+        preamble.append(contentsOf: newPreamble)
+      } catch {
+        context.addDiagnostics(from: error, node: attribute)
+      }
+    }
+
+    return preamble
+  }
+
+  private func expandBody(
```

Comment:
Since this expands both body and preamble, I would call the function `expandBodyAndPreamble`.